src/mame/vsystem/fromanc2.cpp

```diff
@@ -45,23 +45,25 @@ void fromanc2_state::sndcmd_w(uint16_t data)
 	m_sndcpu_nmi_flag = 0;
 }
 
-void fromanc2_state::portselect_w(uint16_t data)
+void fromanc2_base_state::portselect_w(uint16_t data)
 {
 	m_portselect = data;
 }
 
-uint16_t fromanc2_state::keymatrix_r()
+uint16_t fromanc2_base_state::keymatrix_r()
 {
 	uint16_t ret;
 
 	switch (m_portselect)
 	{
-		case 0x01:  ret = ioport("KEY0")->read(); break;
-		case 0x02:  ret = ioport("KEY1")->read(); break;
-		case 0x04:  ret = ioport("KEY2")->read(); break;
-		case 0x08:  ret = ioport("KEY3")->read(); break;
-		default:    ret = 0xffff;
-			logerror("PC:%08X unknown %02X\n", m_maincpu->pc(), m_portselect);
+		case 0x01:  ret = m_in_key[0]->read(); break;
+		case 0x02:  ret = m_in_key[1]->read(); break;
+		case 0x04:  ret = m_in_key[2]->read(); break;
+		case 0x08:  ret = m_in_key[3]->read(); break;
+		default:
+			ret = 0xffff;
+			if (!machine().side_effects_disabled())
+				logerror("PC:%08X unknown %02X\n", m_maincpu->pc(), m_portselect);
 			break;
 	}
```

Comment:
This is a key matrix that the author failed to recognise.  It’s better to emulate them properly.